kratos/containers/distributed_sparse_graph.h

```diff
@@ -0,0 +1,458 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Riccardo Rossi
+//
+
+#if !defined(KRATOS_DISTRIBUTED_SPARSE_GRAPH_H_INCLUDED )
+#define  KRATOS_DISTRIBUTED_SPARSE_GRAPH_H_INCLUDED
+
+
+// System includes
+#include <iostream>
+#include "includes/ublas_interface.h"
+#include "includes/serializer.h"
+#include "includes/parallel_environment.h"
+#include "containers/distributed_numbering.h"
+#include "utilities/communication_coloring_utilities.h"
+#include "containers/sparse_graph.h"
+#include "containers/sparse_contiguous_row_graph.h"
+#include "utilities/parallel_utilities.h"
+
+// External includes
+#include <unordered_map>
+#include <unordered_set>
+
+// Project includes
+#include "includes/define.h"
+
+
+namespace Kratos
+{
+///@addtogroup ApplicationNameApplication
+///@{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// Short class definition.
+//class to construct and store a matrix graph. Can be used to construct efficiently a CSR matrix (or other sparse matrix types)
+
+/** This class is designed to store a matrix graph, aimed at the fast construction of other
+ * sparse matrix formats (particularly CSR)
+ * IMPORTANT NOTE: it is BY DESIGN NOT threadsafe! (a graph should be computed in each thread and then merged)
+*/
+template< class TIndexType >
+class DistributedSparseGraph //: public SparseGraph
+{
+public:
+    ///@name Type Definitions
+    ///@{
+    typedef TIndexType IndexType; //note that this could be different from the one in the basetype
```

Comment:
I think that the comment is a leftover from a previous implementation.

Reply:
ok